opentelemetry-otlp/src/lib.rs

```diff
@@ -408,3 +405,9 @@ pub enum Protocol {
     /// HTTP protocol with binary protobuf
     HttpBinary,
 }
+
+#[derive(Debug, Default)]
+#[doc(hidden)]
+/// Placeholder type when no exporter pipeline has been configured in telemetry pipeline.
+/// Users shouldn't be able to use this type, hence the private PhantomData.
+pub struct NoExporterConfig(PhantomData<()>);
```

Comment:
nit: This doesn't require `PhantomData` as ` NoExporterConfig(());` also has a private constructor